src/GridCal/Engine/Core/DataStructures/generator_data.py

```diff
@@ -45,6 +45,16 @@ def __init__(self, ngen, nbus, ntime=1):
 
         self.C_bus_gen = sp.lil_matrix((nbus, ngen), dtype=int)
 
+        # r0, r1, r2, x0, x1, x2
+        self.generator_r0 = np.zeros(ngen)
```

Comment:
All the new properties also need to be included in the slice function

Reply:
Ok, added in the new commit